code/_onclick/hud/human.dm

```diff
@@ -87,7 +87,10 @@
 
 /datum/hud/human/New(mob/living/carbon/human/owner, ui_style = 'icons/mob/screen_midnight.dmi')
 	..()
-	owner.overlay_fullscreen("see_through_darkness", /obj/screen/fullscreen/see_through_darkness)
+	if("Night Vision" in owner.client.prefs.positive_traits) //check prefs directly due to the order in which things are made

```

Comment:
🤢

Reply:
HUDs are created before traits are assigned, and I'm not sure of a better way to do this. If you have any advice, feel free to help out.

Reply:
consider the possibility that traits may be lost midround one day, as part of a random event or a surgery or whatever.

you're checking whether they should have this trait based on if they WANT it, not if they have it, that's the secondary puke-y thing about this code.

the first one is that it's a string in list check, very bad. Either, change the feature up slightly so that it instead increases the human's eye organ's ```see_in_dark``` (just a ++ will do) or make this nightvision/see_through_dark object be a var on the person's eyes, and make receiving a set of eyes regenerate this part of your hud (shouldn't be difficult)

Reply:
I thought traits were datums, why are you doing string lists?

Reply:
The string list is used to track names in prefs with the ``SStraits` lists. When I wrote the code it was faster and cleaner.